internal/data/metric.go

```diff
@@ -49,6 +50,17 @@ func NewMetricData() MetricData {
 	return MetricData{&orig}
 }
 
+// Clone returns a copy of MetricData.
+func (md MetricData) Clone() MetricData {
+	otlp := MetricDataToOtlp(md)
+	resourceMetricsClones := make([]*otlpmetrics.ResourceMetrics, 0, len(otlp))
+	for _, resourceMetrics := range otlp {
+		resourceMetricsClones = append(resourceMetricsClones,
+			proto.Clone(resourceMetrics).(*otlpmetrics.ResourceMetrics))
+	}
+	return MetricDataFromOtlp(resourceMetricsClones)
+}
+
```

Comment:
Some tests for clone would be good. Pass an otlp check that Clone returns equalValues but not same pointer.

Reply:
it's being tested, but not this function directly https://github.com/open-telemetry/opentelemetry-collector/pull/717/files#diff-50214c26172368731479968d001a4102R222